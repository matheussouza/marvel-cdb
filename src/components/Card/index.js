import React from 'react';

const Card = ({ imagePath, name }) => {
  return (
    <div className="card text-center" style={{ marginBottom: 20 }}>
      <img src={imagePath} className="card-img-top" alt={name} />
      <div className="card-body" style={{ padding: 8, minHeight: 76, display: 'flex', alignItems: 'center', justifyContent: 'center' }}>
        <h5 style={{ margin: 0 }} className="card-title">{name}</h5>
      </div>
    </div>
  )
}

export default Card;
