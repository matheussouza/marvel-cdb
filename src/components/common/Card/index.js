import React from 'react';
import defaultPath from '../../../assets/img/landscape_not_available.jpg';

export default ({ imagePath, name }) => {
  return (
    <div className="card text-center app-card">
      <img src={imagePath || defaultPath} className="card-img-top" alt={name} />
      <div className="card-body app-card-body">
        <h5 className="card-title app-container-reset">{name}</h5>
      </div>
    </div>
  )
}
