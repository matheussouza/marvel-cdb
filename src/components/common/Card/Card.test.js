import React from 'react';
import { render } from '@testing-library/react';

import Card from './index';

describe('Card component', () => {

  it('should render', () => {
    const subject = render(<Card />)
    expect(subject.container).toMatchSnapshot();
  });

  it('should render with name Spider-Man', () => {
    const subject = render(<Card name="Spider-Man" />);

    expect(subject.getByRole('img')).toHaveProperty('alt' ,'Spider-Man');
    expect(subject.getByRole('heading').innerHTML).toMatch('Spider-Man');
  });

  it('should render default image missing imagePath', () => {
    const subject = render(<Card />);

    expect(subject.getByRole('img')).toHaveProperty('src', 'http://localhost/landscape_not_available.jpg');
  });

  it('should render imagePath', () => {
    const subject = render(<Card imagePath="http://anyimagepath.com/image_path.jpg" />);

    expect(subject.getByRole('img')).toHaveProperty('src', 'http://anyimagepath.com/image_path.jpg');
  });

});