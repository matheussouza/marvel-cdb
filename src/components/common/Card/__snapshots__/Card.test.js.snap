// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render 1`] = `
<div>
  <div
    class="card text-center app-card"
  >
    <img
      class="card-img-top"
      src="landscape_not_available.jpg"
    />
    <div
      class="card-body app-card-body"
    >
      <h5
        class="card-title app-container-reset"
      />
    </div>
  </div>
</div>
`;
